test/SlowTests/Client/TimeSeries/Query/TimeSeriesQueryTests.cs

```diff
@@ -1189,17 +1243,3913 @@ as Stocks
+            using (var store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                using (var session = store.OpenSession())
+                {
+                    for (int i = 1; i <= 3; i++)
+                    {
+                        var id = $"people/{i}";
+
+                        session.Store(new Person
+                        {
+                            Name = "Oren",
+                            Age = i * 30
+                        }, id);
+
+                        var tsf = session.TimeSeriesFor(id);
+
+                        tsf.Append("HeartRate", baseline.AddMinutes(61), "watches/fitbit", new[] { 59d, 141.5 });
+                        tsf.Append("HeartRate", baseline.AddMinutes(62), "watches/fitbit", new[] { 79d, 142.82 });
+                        tsf.Append("HeartRate", baseline.AddMinutes(63), "watches/apple", new[] { 69d });
+
+                        tsf.Append("HeartRate", baseline.AddMonths(1).AddMinutes(61), "watches/apple", new[] { 159d, 138.12 });
+                        tsf.Append("HeartRate", baseline.AddMonths(1).AddMinutes(62), "watches/fitbit", new[] { 179d });
+                        tsf.Append("HeartRate", baseline.AddMonths(1).AddMinutes(63), "watches/fitbit", new[] { 169d, 142.57 });
+                    }
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = session.Advanced.RawQuery<TimeSeriesAggregation>(@"
+declare timeseries out(x) 
+{
+    from x.HeartRate between $start and $end
+        where Values[1] != null AND NOT Tag = 'watches/apple'
+    group by '1 month' 
+    select min(), max(), avg()
+}
+from People as doc
+where doc.Age > 49
+select out(doc)
+")
+                        .AddParameter("start", baseline)
+                        .AddParameter("end", baseline.AddMonths(2));
+                    var result = query.ToList();
+
+                    Assert.Equal(2, result.Count);
+
+                    for (int i = 0; i < 2; i++)
+                    {
+                        var agg = result[i];
+
+                        Assert.Equal(3, agg.Count);
+
+                        Assert.Equal(2, agg.Results.Length);
+
+                        var val = agg.Results[0];
+
+                        Assert.Equal(59, val.Min[0]);
+                        Assert.Equal(79, val.Max[0]);
+                        Assert.Equal(69, val.Avg[0]);
+                        Assert.Equal(2, val.Count[0]);
+
+                        Assert.Equal(141.5, val.Min[1]);
+                        Assert.Equal(142.82, val.Max[1]);
+                        Assert.Equal(2, val.Count[1]);
+                        Assert.Equal((141.5 + 142.82) / 2, val.Avg[1]);
+
+                        var expectedFrom = new DateTime(baseline.Year, baseline.Month, 1, 0, 0, 0);
+                        var expectedTo = expectedFrom.AddMonths(1);
+
+    
```

Comment:
Can I also write:

```
declare timeseries out(x) 
{
    from x.HeartRate between $start and $end
        load $config as src
        where (Values[0] between src.Min and src.Max) OR 
              (src.IsoCompliant = true AND Values[0] > $val)  
    group by '1 month' 
    select min(), max(), avg()
}
```

Where the `$config` is some arbitrary document?

Reply:
no, this is not supported, we can only load by Tag
do we want to support that?

we can achieve the same by loading the document in root RQL and 
pass it as an argument to the timeseries function, no? 
e.g.
```
declare timeseries out(x, src) 
{
    from x.HeartRate between $start and $end
        where (Values[0] between src.Min and src.Max) OR 
              (src.IsoCompliant = true AND Values[0] > $val)  
    group by '1 month' 
    select min(), max(), avg()
}
from users as u
load $config as src
select out(u, src)
```